controller/workitem_blackbox_test.go

```diff
@@ -1659,6 +1660,170 @@ func (s *WorkItem2Suite) TestWI2ListByStateFilterOKModifiedUsingIfNoneMatchIfMod
 }
 
+func (s *WorkItem2Suite) TestListOrder() {
+	s.T().Run("list with explicit order", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, s.DB,
+			tf.Iterations(2),
+			tf.WorkItems(10, func(fxt *tf.TestFixture, idx int) error {
+				switch idx {
+				case 0, 1, 2, 3, 4, 5, 6:
+					fxt.WorkItems[idx].Fields[workitem.SystemState] = "open"
+				default:
+					fxt.WorkItems[idx].Fields[workitem.SystemState] = "new"
+				}
+				return nil
+			}),
+		)
+
+		t.Run("by created descending", func(t *testing.T) {
+			// when
+			exp := ptr.String(`{"system.state": "open"}`)
+			sort := ptr.String("-created")
+			_, actualWIs := test.ListWorkitemsOK(s.T(), s.svc.Context, s.svc, s.workitemsCtrl, fxt.Spaces[0].ID, exp, nil, nil, nil, nil, nil, nil, nil, nil, nil, sort, nil, nil)
+			// then
+			require.NotNil(s.T(), actualWIs)
+			require.Len(s.T(), actualWIs.Data, 7)
+			toBeFound := id.Slice{
+				fxt.WorkItems[0].ID,
+				fxt.WorkItems[1].ID,
+				fxt.WorkItems[2].ID,
+				fxt.WorkItems[3].ID,
+				fxt.WorkItems[4].ID,
+				fxt.WorkItems[5].ID,
+				fxt.WorkItems[6].ID,
+			}.ToMap()
+
+			for i := 0; i <= 6; i++ {
+				require.Equal(t, fxt.WorkItems[i].ID, *actualWIs.Data[6-i].ID)
+			}
+
+			for _, wi := range actualWIs.Data {
+				_, ok := toBeFound[*wi.ID]
+				require.True(t, ok, "unknown work item found: %s", wi.ID)
+				delete(toBeFound, *wi.ID)
+			}
+			require.Empty(t, toBeFound, "failed to find all work items: %+s", toBeFound)
+		})
+		t.Run("by created ascending", func(t *testing.T) {
+			exp := ptr.String(`{"system.state": "open"}`)
+			sort := ptr.String("created")
+			_, actualWIs := test.ListWorkitemsOK(s.T(), s.svc.Context, s.svc, s.workitemsCtrl, fxt.Spaces[0].ID, exp, nil, nil, nil, nil, nil, nil, nil, nil, nil, sort, nil, nil)
+			// then
+			require.NotNil(s.T(), actualWIs)
+			require.Len(s.T(), actualWIs.Data, 7)
+			toBeFound := id.Slice{
+				fxt.WorkItems[0].ID,
+				fxt.WorkItems[1].ID,
+				fxt.WorkItems[2].ID,
+				fxt.WorkItems[3].ID,
+				fxt.WorkItems[4].ID,
+				fxt.WorkItems[5].ID,
+				fxt.WorkItems[6].ID,
+			}.ToMap()
+
+			for i := 0; i <= 6; i++ {
+				require.Equal(t, fxt.WorkItems[i].ID, *actualWIs.Data[i].ID)
+			}
+
+			for _, wi := range actualWIs.Data {
+				_, ok := toBeFound[*wi.ID]
+				require.True(t, ok, "unknown work item found: %s", wi.ID)
+				delete(toBeFound, *wi.ID)
+			}
+			require.Empty(t, toBeFound, "failed to found all work items: %+s", toBeFound)
+		})
+
+		t.Run("by updated descending", func(t *testing.T) {
+			// when
+			for _, v := range []int{3, 2, 0, 1, 6, 5, 4} {
+				payload := app.UpdateWorkitemPayload{
+					Data: &app.WorkItem{
+						Type: APIStringTypeWorkItem,
+						ID:   &fxt.WorkItems[v].ID,
+						Attributes: map[string]interface{}{
+							workitem.SystemState:   "resolved",
+							workitem.SystemVersion: fxt.WorkItems[v].Version,
+						},
+					},
+				}
+
+				test.UpdateWorkitemOK(s.T(), s.svc.Context, s.svc, s.workitemCtrl, fxt.WorkItems[3].ID, &payload)
```

Comment:
Why are you updating only `fxt.WorkItems[3]` here? Shouldn't it update all 3,2,0,1,6,5 and 4 indexed workitems?

Reply:
Fixed in 902561db05682910b875a49091a397a1f86ef227